src/components/PlanningTable/index.tsx

```diff
@@ -61,6 +54,55 @@ export const PlanningTable = <TData, TValue>({
     getFacetedUniqueValues: getFacetedUniqueValues()
   })
 
+  // Handle navigation using arrow keys
+  useEffect(() => {
+    if (!onRowSelected) {
+      return
+    }
+
+    const keyDownHandler = (evt: KeyboardEvent): void => {
+      if (!table || !['ArrowDown', 'ArrowUp', 'Escape'].includes(evt.key)) {
+        return
+      }
+
+      evt.preventDefault()
+      const rows = table.getRowModel().rows
+      if (!rows?.length) {
+        return
+      }
+
+      const selectedRows = table.getSelectedRowModel()
+      const selectedRow = selectedRows?.rows[0]
+
+      if (evt.key === 'Escape') {
+        if (selectedRow) {
+          selectedRow.toggleSelected(false)
+        }
+      } else if (!selectedRow) {
+        const idx = evt.key === 'ArrowDown' ? 0 : rows.length - 1
+        rows[idx].toggleSelected(true)
+      } else {
+        const nextIdx = selectedRow.index + ((evt.key === 'ArrowDown') ? 1 : -1)
+        const idx = nextIdx < 0 ? rows.length - 1 : nextIdx >= rows.length ? 0 : nextIdx
```

Comment:
Maybe this could be a function? Nested ternaries are a bit hard to read.
Maybe do something about if-if-else-if-else-chunk as well?

Reply:
Agree that it is not perfect. Will leave it unchanged as we will refactor this when we have refactored NavigationWrapper into ViewWrapper and then add hooks to simplify this.